app/src/main/java/at/bitfire/icsdroid/model/EditSubscriptionModel.kt

```diff
@@ -82,8 +90,20 @@ class EditSubscriptionModel(
 
                 // notify UI about success
                 uiState = uiState.copy(successMessage = getApplication<Application>().getString(R.string.edit_calendar_deleted))
-            }
+            } ?: Log.w(Constants.TAG, "There's no subscription to remove")
         }
     }
 
+    fun shareUrl(activityContext: Context) = viewModelScope.launch {
```

Comment:
I would say Intent handling and launching is to be kept inside the activity (one of the exceptions). This is for two reasons:
- because they "logically" are about navigating the app, hence they might be replaced with the [NavController](https://developer.android.com/reference/androidx/navigation/NavController#handleDeepLink(android.content.Intent)) in the future 
- because they require the acitivity context

Reply:
I agree, I will just use the activity's scope